[Safer CPP] Address issues in WebGLRenderingContext

roberto-apple opened the pull request:
#### 4a378decc450093b8fb2450dff28c38532d92be5
<pre>
[Safer CPP] Address issues in WebGLRenderingContext
<a href="https://bugs.webkit.org/show_bug.cgi?id=293100">https://bugs.webkit.org/show_bug.cgi?id=293100</a>
<a href="https://rdar.apple.com/151439889">rdar://151439889</a>

Reviewed by Chris Dumez.

Address all outstanding Safer CPP issues in WebGLRenderingContext.

* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/html/canvas/WebGLRenderingContext.cpp:
(WebCore::WebGLRenderingContext::getExtension):
(WebCore::WebGLRenderingContext::getSupportedExtensions):
(WebCore::WebGLRenderingContext::getFramebufferAttachmentParameter):
(WebCore::WebGLRenderingContext::getInt64Parameter):
(WebCore::WebGLRenderingContext::maxDrawBuffers):
(WebCore::WebGLRenderingContext::maxColorAttachments):
(WebCore::WebGLRenderingContext::addMembersToOpaqueRoots):

Canonical link: <a href="https://commits.webkit.org/295215@main">https://commits.webkit.org/295215@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d83181fe162509feb7bffdc39f27fd1e44dd3441

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104310 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/24014 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14428 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109511 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54978 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/106350 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24386 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32561 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79214 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107316 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18956 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94127 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59543 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/18826 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/12195 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54338 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/88459 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12253 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111892 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31467 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23183 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88240 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31831 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90367 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87903 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22407 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32802 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10559 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/25935 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31395 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/36722 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31189 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34525 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/32749 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/html/canvas/WebGLRenderingContext.cpp:
```diff
@@ -135,7 +135,8 @@ std::optional<WebGLExtensionAny> WebGLRenderingContext::getExtension(const Strin
 
     // When adding extensions that use enableDraftExtensions, add them to the webgl-draft-extensions-flag.js test.
-    [[maybe_unused]] const bool enableDraftExtensions = scriptExecutionContext()->settingsValues().webGLDraftExtensionsEnabled;
+    RefPtr scriptExecutionCtx = scriptExecutionContext();
```
Abbreviations such as "scriptExecutionCtx" are not OK, per WebKit coding style.

roberto-apple replied:
Corrected

cdumez commented on Source/WebCore/html/canvas/WebGLRenderingContext.cpp:
```diff
@@ -318,17 +323,17 @@ WebGLAny WebGLRenderingContext::getFramebufferAttachmentParameter(GCGLenum targe
         return 0;
     if (!m_maxDrawBuffers)
-        m_maxDrawBuffers = m_context->getInteger(GraphicsContextGL::MAX_DRAW_BUFFERS_EXT);
+        m_maxDrawBuffers = protectedGraphicsContextGL()->getInteger(GraphicsContextGL::MAX_DRAW_BUFFERS_EXT);
```
We should use a local variable to avoid repeated calls to `protectedGraphicsContextGL()` in this function.

roberto-apple replied:
Corrected

cdumez commented on Source/WebCore/html/canvas/WebGLRenderingContext.cpp:
```diff
@@ -347,7 +352,8 @@ void WebGLRenderingContext::addMembersToOpaqueRoots(JSC::AbstractSlotVisitor& vi
 
     Locker locker { objectGraphLock() };
-    addWebCoreOpaqueRoot(visitor, m_activeQuery.get());
+    RefPtr activeQuery = m_activeQuery.get();
```
Unfortunate to introduce a local variable just for this, we could use:
```
addWebCoreOpaqueRoot(visitor, RefPtr { m_activeQuery.get() }.get());
```

roberto-apple replied:
Corrected

cdumez commented on Source/WebCore/html/canvas/WebGLRenderingContext.cpp:
```diff
@@ -148,46 +149,47 @@ std::optional<WebGLExtensionAny> WebGLRenderingContext::getExtension(const Strin
-    ENABLE_IF_REQUESTED(WebGLCompressedTexturePVRTC, m_webglCompressedTexturePVRTC, "WEBKIT_WEBGL_compressed_texture_pvrtc", WebGLCompressedTexturePVRTC::supported(*m_context));
-    ENABLE_IF_REQUESTED(WebGLCompressedTextureS3TC, m_webglCompressedTextureS3TC, "WEBGL_compressed_texture_s3tc", WebGLCompressedTextureS3TC::supported(*m_context));
-    ENABLE_IF_REQUESTED(WebGLCompressedTextureS3TCsRGB, m_webglCompressedTextureS3TCsRGB, "WEBGL_compressed_texture_s3tc_srgb", WebGLCompressedTextureS3TCsRGB::supported(*m_context));
+    RefPtr graphicsContext = protectedGraphicsContextGL();
```
We don't use `protectedFoo()` functions to initialize local variables. It is more verbose and has no benefits. `protectedFoo()` are meant to be used to avoid local variables.

Here you should probably use:
```
RefPtr graphicsContext = graphicsContextGL();
```

roberto-apple replied:
Ah! Of course. 
Corrected

cdumez reviewed (approved):
LGTM with fix.

roberto-apple committed:
[Safer CPP] Address issues in WebGLRenderingContext
https://bugs.webkit.org/show_bug.cgi?id=293100
rdar://151439889

Reviewed by Chris Dumez.

Address all outstanding Safer CPP issues in WebGLRenderingContext.

* Source/WebCore/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebCore/html/canvas/WebGLRenderingContext.cpp:
(WebCore::WebGLRenderingContext::getExtension):
(WebCore::WebGLRenderingContext::getSupportedExtensions):
(WebCore::WebGLRenderingContext::getFramebufferAttachmentParameter):
(WebCore::WebGLRenderingContext::getInt64Parameter):
(WebCore::WebGLRenderingContext::maxDrawBuffers):
(WebCore::WebGLRenderingContext::maxColorAttachments):
(WebCore::WebGLRenderingContext::addMembersToOpaqueRoots):

Canonical link: https://commits.webkit.org/295215@main
